src/core/builders/create-form-validator.spec.ts

```diff
@@ -847,4 +847,122 @@ describe("createFormValidator", () => {
       { field: Schema.arrayObjectString.nth(3), error: "INVALID_VALUE" },
     ]);
   });
+
+  it("dependency change should trigger validation run", async () => {
```

Comment:
I think you forgot about other half of this feature :D 

the value of dependency should be passed to the function which returns list of rules
`(dep1, dep2) => [rule1, rule2]` - thats the point of having dependencies